antarest/study/storage/variantstudy/snapshot_generator.py

```diff
@@ -77,22 +77,39 @@ def generate_snapshot(
         assert_permission_on_studies(jwt_user, [root_study, *descendants], StudyPermissionType.READ, raising=True)
         ref_study, cmd_blocks = search_ref_study(root_study, descendants, from_scratch=from_scratch)
 
-        # We are going to generate the snapshot in a temporary directory which will be renamed
-        # at the end of the process. This prevents incomplete snapshots in case of error.
-
-        # Get snapshot directory and prepare a temporary directory next to it.
+        # Get snapshot directory
         variant_study = descendants[-1]
         snapshot_dir = variant_study.snapshot_dir
-        snapshot_dir.parent.mkdir(parents=True, exist_ok=True)
-        self._tmp_dir = Path(tempfile.mkdtemp(dir=snapshot_dir.parent, prefix=f"~{snapshot_dir.name}", suffix=".tmp"))
+
+        if snapshot_dir.exists():
```

Comment:
Also I think we have an issue in `search_ref_study` at lines 281-284:
```python
command_ids = [c.id for c in ref_study.commands]
last_exec_index = command_ids.index(last_exec_cmd)
cmd_blocks = ref_study.commands[last_exec_index + 1 :]
```
If the last executed command is not anymore in the list of commands (it has been deleted), there will probably be an exception or at least an unexpected behaviour.

Reply:
I changed the `search_ref_study` so that it returns a `force_regenerate` flag to force the snapshot regeneration.
This flag should fix the issues.
But, we need to discuss about "if snapshots are identified as "old" in search_ref_study, we will be in the same situation"...